photon-server/src/main/java/org/photonvision/server/RequestHandler.java

```diff
@@ -608,6 +655,114 @@ public static void onImportObjectDetectionModelRequest(Context ctx) {
                                 UIPhotonConfiguration.programStateToUi(ConfigManager.getInstance().getConfig())));
     }
 
+    public static void onExportObjectDetectionModelRequest(Context ctx) {
+        // TODO: implement this
+    }
+
+    public static void onDeleteObjectDetectionModelRequest(Context ctx) {
+        // TODO: implement this
+        logger.info("Deleting object detection model");
+        Path modelPath;
+
+        try {
+            modelPath = Path.of(kObjectMapper.readTree(ctx.bodyInputStream()).get("modelPath").asText());
+
+            if (modelPath == null) {
+                ctx.status(400);
+                ctx.result("The provided model path was malformed");
+                logger.error("The provided model path was malformed");
+                return;
+            }
+
+            if (!modelPath.toFile().exists()) {
+                ctx.status(400);
+                ctx.result("The provided model path does not exist");
+                logger.error("The provided model path does not exist");
+                return;
+            }
+
+            if (!modelPath.toFile().delete()) {
+                ctx.status(500);
+                ctx.result("Unable to delete the model file");
+                logger.error("Unable to delete the model file");
+                return;
+            }
+
+            if (!ConfigManager.getInstance()
+                    .getConfig()
+                    .getNeuralNetworkProperties()
+                    .removeModel(modelPath)) {
+                ctx.status(400);
+                ctx.result("The model's information was not found in the config");
+                logger.error("The model's information was not found in the config");
+                return;
+            }
+
+            NeuralNetworkModelManager.getInstance().discoverModels();
+
+            ctx.status(200).result("Successfully deleted object detection model");
+
+        } catch (Exception e) {
+            ctx.status(500);
+            ctx.result("Error deleting object detection model: " + e.getMessage());
+            logger.error("Error deleting object detection model", e);
+        }
+
+        DataChangeService.getInstance()
+                .publishEvent(
+                        new OutgoingUIEvent<>(
+                                "fullsettings",
+                                UIPhotonConfiguration.programStateToUi(ConfigManager.getInstance().getConfig())));
+    }
+
+    public static void onRenameObjectDetectionModelRequest(Context ctx) {
+        try {
+            var data = kObjectMapper.readTree(ctx.bodyInputStream());
```

Comment:
Prefer a record (effectively a named tuple) over `readTree` where possible to statically type this.